executor/src/subscriber.rs

```diff
@@ -157,17 +187,29 @@ impl Subscriber {
     /// sequenced we will not quit this method until we have successfully
     /// fetched the payload.
     async fn wait_on_payload(
```

Comment:
It seems possible that parents of `certificate_id` may not exist in the certificate store, and this function triggers block waiter to fetch the parent certificates and the certificate itself as well, before fetching their payloads. If my understanding is correct, maybe we can update the function comment?

Reply:
I don't think this is possible. Consensus would fail if we don't have all the ancestors of a certificate that we try to sequence. The role of the `certificate-waiter` is exactly to prevent that.

Reply:
I see. If the `GetBlock` message here never triggers causal completion (because the parents must already be in the certificate store), which logic triggers the causal completion?

Reply:
In `certificate waiter` I only see logic to wait for parents to become available, but not the logic to trigger fetching parents. Maybe I'm missing something ..

Reply:
@mwtian if this is still unclear to you let's catch up offline, I can explain the process

Reply:
Great, thanks! We can discuss offline.